File: centrum/C/tudakozo.c
//Gyakorló könvtári modul
#include <stdio.h>
#include <stdlib.h>
#include "tudakozo.h"
#define true 1
#define false 0

int _N,_C,_K;
int _kerd=0;
int _Init=true;

int hany(){
  if (_Init) {
    scanf("%d %d %d",&_N, &_C, &_K);
    _Init = false;
    _kerd=0;
  }
  return _N;
} /*hany*/

int kerdes(int x, int y){
  if(_Init){
    printf("HIBA: előbb hívd a hany műveletet!\n");
    exit(1);
  }
  _kerd++;
  if(_kerd>3*_N){
    printf("HIBA: túl sok kérdés volt!\n");
    exit(1);
  }
  if(x<1 || x>_N || y<1 || y>_N){
    printf("HIBA: hibás sorszám!\n");
    exit(1);
  }
  if (x==_K && y!=_C || y==_K && x!=_C){
    return 0;
  }else
    return 1;
}
void megoldas(int m){
//Itt ellenörzik, hogy a feltett kérdések apalján m lehet-e centrum
  if (m<1 ||m>_N){
    printf("HIBA: hibás sorszám\n");
    exit(1);
  }
  if(m==_C)
    if (_kerd<=2*_N)
      printf("HELYES, centrum= %d kérdésszám= %d <=2*N\n", m,_kerd);
    else
      printf("HELYES, centrum= %d kérdésszám= %d <=3*N\n", m,_kerd);
  else
    printf("HIBÁS\n");
  exit(0);
}

File: centrum/Pas/tudakozo.pas
unit tudakozo;
//gyakorló könyvtári modul
interface
  function hany:integer;
  function kerdes(x, y:integer):integer;
  procedure megoldas(m:integer);

implementation
  const
    Init:boolean=true;
var
    N,C,K:integer;
    kerd:integer;
function hany:integer;
begin
	readln(N,C,K);
    kerd:=0;
    Init:=false;
    hany:=N;
end{hany};

function kerdes(x,y:integer):integer;
begin
	if(Init) then begin
		writeln('HIBA: Előbb a hany műveletet kell hívni!');
		halt;
	end;
	inc(kerd);
	if (kerd>3*N )then begin
		writeln('HIBA: túl sok kérdés volt már!');
		halt;
	end;
	if(x<1)or(x>N)or(y<1)or(y>N)then begin
		writeln('HIBA: hibás sorszám!',x,'::',y);
		halt;
	end;
	if (x=K)and(y<>C)or(y=K)and(x<>C) then
		kerdes:=0
	else
		kerdes:=1;
end{kerdes};
procedure megoldas(m:integer);
begin
{Itt ellenőrzik, hogy a feltett kérdések alapján m lehet-e megoldás!}
	if (m<1) or(m>N)then begin
		writeln('HIBA: hibás sorszám');
		halt;
	end;
	if (m=C) then begin
		if kerd<=2*N then
			writeln('HELYES, centrum=',m,' kérdesszám = ',kerd,' <=2*N')
		else
			writeln('HELYES, centrum=',m,' kérdesszám = ',kerd,' <=3*N');
	end else 
		writeln('HIBÁS');
	halt;
end{megoldas};

begin

end.
